modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java

```diff
@@ -414,33 +435,30 @@ public BigInteger getTotalDifficulty() {
             long maxNumber = getMaxNumber();
 
             List<BlockInfo> blockInfos = index.get(maxNumber);
-            for (BlockInfo blockInfo : blockInfos) {
-                if (blockInfo.isMainChain()) {
-                    return blockInfo.getCummDifficulty();
-                }
-            }
-
-            while (true) {
-                --maxNumber;
```

Comment:
The reason for removing these code because the logic should not inside the gettotalDiffcalty. Also, it does not fix anything for the corrupt index DB. Therefore I put the else case to indicate the error.

Reply:
Revised the method implementation. It does check the previous blockInfo but only 128 depth.